### PR TITLE
Add installation module for custom kernel flavor with kernel-64kb support

### PR DESCRIPTION
**Pull request by czerw**
Enhancement poo#151153: We should to be able to install any kernel
flavor. New variable KERNEL_FLAVOR was introduced. 

First  use case is for kernel-64kb on aarch64 baremetal.

- Related ticket: https://progress.opensuse.org/issues/151153
- Needles:  NONE

##### KERNEL_FLAVOR=kernel-64kb
15-SP5 installation: https://openqa.suse.de/tests/12879693
15-SP5 install_ltp with update_ltp for KOTD: https://openqa.suse.de/tests/12879694 
15-SP5 cve: https://openqa.suse.de/tests/12881277
15-SP5 syscalls: https://openqa.suse.de/tests/12881278

##### Without KERNEL_FLAVOR
15-SP5 installation: https://openqa.suse.de/tests/12881505
15-SP5 install_ltp with update_ltp for KOTD: https://openqa.suse.de/tests/12881506
15-SP5 cve: https://openqa.suse.de/tests/12881507
15-SP5 syscalls: https://openqa.suse.de/tests/12881508

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -2351,6 +2351,7 @@ sub load_system_prepare_tests {
     }
     loadtest 'console/integration_services' if is_hyperv || is_vmware;
     loadtest 'console/hostname' unless is_bridged_networking;
+    loadtest 'kernel/install_kernel_flavor' if get_var('KERNEL_FLAVOR');
```
Comment (czerw) [inline lib/main_common.pm:2354]: I'm aware about similar approach for RT. RT will be redone and migrated to `KERNEL_FLAVOR` as well in separate change.

File: tests/kernel/install_kernel_flavor.pm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils qw(zypper_call);
+
+sub run {
+    select_serial_terminal;
+
+    my $kernel_flavor = get_var('KERNEL_FLAVOR');
```
Comment (pevik) [inline tests/kernel/install_kernel_flavor.pm:19]: maybe `get_required_var('KERNEL_FLAVOR');` and then avoid if in `zypper_call` line?

File: lib/LTP/utils.pm [hunk 3]
```diff
@@ -402,6 +403,10 @@ sub get_default_pkg {
 }
 
 sub install_from_repo {
+    # Workaround for kernel-64kb, until we add multibuild support to LTP package
+    # Lock kernel-default to don't pull it as LTP dependency
+    zypper_call 'al kernel-default' if get_kernel_flavor eq 'kernel-64kb';
```
Comment (mdoucha) [inline lib/LTP/utils.pm:408]: Would it make sense to lock `kernel-default` for all non-default flavors?

Comment (czerw) [reply lib/LTP/utils.pm:408]: Rather not at the moment, as current approach worked fine for other kernels. I will include multibuild support to LTP to cleanup work which I already planned https://progress.opensuse.org/issues/132182.

File: lib/main_common.pm (hunk 1 above)
Comment (czerw) [reply lib/main_common.pm:2354]: RT part will be migrated  in https://progress.opensuse.org/issues/132542

File: lib/LTP/utils.pm (hunk 3 above)
Comment (mdoucha) [reply lib/LTP/utils.pm:408]: OK

**Review by mdoucha — APPROVED** [review]

**Pull request merged**